### PR TITLE
PLATFORM-3183: Ensure all AJAX requests in MediaWiki are called over the correct protocol

### PR DESCRIPTION
**Pull request by themech**
Prepare ajax calls to work when the page is requested over both: http and https protocols.
If we don't specify the server in the path, it should work for each protocol. Also we probably don't need the server name as we're not making cross-domain calls here.

### CODE REVIEW

**Comment by mszabo-wikia** [comment]
It seems this PR is not up to date with https://github.com/Wikia/app/pull/13884 changes

File: extensions/wikia/AdEngine/tests/AdEngine2ContextServiceTest.php [hunk 1]
```diff
@@ -230,6 +230,7 @@ public function testGetContext(
 		$customDartKvs = 'a=b;c=d';
 		$catId = WikiFactoryHub::CATEGORY_ID_LIFESTYLE;
 		$shortCat = 'shortcat';
+		// @todo(mech); change this once $wgEnableLocalResourceLoaderLinks is enabled globally
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/tests/AdEngine2ContextServiceTest.php]: INFO Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by themech** [comment]
@TK-999 true, I didn't not touch chat. But if I understand correctly, chat https://github.com/Wikia/app/pull/13884 is about links, whereas here I'm just touching ajax requests sent to MW. For now, we're not touching any links to articles or other pages, as this would point users to https versions of pages, which don't work yet.

**Comment by mszabo-wikia** [comment]
Yeah, that's odd. Probably I saw cached diff

**Review by harnash — APPROVED** [review]

**Review by Grunny — COMMENTED** [review]
I think it might be worth being more consistent and using `wgScriptPath` for all of these, not just those that already had it. It should be the same as `wgScriptPath` defaults to and empty string, but it might make things easier later if we were to change the path of these (such as for the language path option) as it could be changed per wiki. What do you think?

**Review by Grunny — APPROVED** [review]

File: extensions/wikia/WikiaMiniUpload/js/WMU.js [hunk 2]
```diff
@@ -932,7 +932,7 @@ function WMU_insertPlaceholder( box ) {
 	WMU_box_filled.push(box);
 	var to_update = $( '#WikiaImagePlaceholder' + box );
 	to_update.html($( '#ImageUploadCode' ).html());
-	$.post(wgServer + wgScript + '?title=' + wgPageName  +'&action=purge');
+	$.post(wgScript + '?title=' + wgPageName  +'&action=purge');
 }
 
 function WMU_insertImage(type) {
```
Comment (WikiaTech) [inline extensions/wikia/WikiaMiniUpload/js/WMU.js:938]: MAJOR Function has a complexity of 27 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[project-id]: wikia:app

File: extensions/wikia/Blogs/js/Blogs.js [hunk 3]
```diff
@@ -146,7 +146,7 @@ Blogs.save = function( event ) {
 	if (oForm.length > 0) {
 		document.body.style.cursor = "wait";
 		var textfield = $( "#blog-comm-textfield-" + id );
```
Comment (WikiaTech) [inline extensions/wikia/Blogs/js/Blogs.js:148]: MAJOR Remove the declaration of the unused 'textfield' variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedVariable)

[project-id]: wikia:app
MAJOR Remove this useless assignment to local variable "textfield" [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)

[project-id]: wikia:app

File: extensions/wikia/SiteWideMessages/SpecialSiteWideMessages.js [hunk 4]
```diff
@@ -34,7 +34,7 @@
 	function SWMAjaxDismiss( e ) {
 		e.preventDefault();
 		var	id = e.data.id,
-			ajaxUrl = wgServer + wgScript + "?title=" + wgPageName + "&action=ajax&rs=SiteWideMessagesAjaxDismiss&rsargs[]=" + id,
+			ajaxUrl = wgScript + "?title=" + wgPageName + "&action=ajax&rs=SiteWideMessagesAjaxDismiss&rsargs[]=" + id,
 			request = $.get(ajaxUrl,function(data){
```
Comment (WikiaTech) [inline extensions/wikia/SiteWideMessages/SpecialSiteWideMessages.js:38]: MAJOR Remove the declaration of the unused 'request' variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedVariable)

[project-id]: wikia:app
MAJOR Remove this useless assignment to local variable "request" [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)

[project-id]: wikia:app
MAJOR Remove the unused function parameter "data". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

[project-id]: wikia:app

File: extensions/wikia/Wall/js/MessageTopic.js [hunk 5]
```diff
@@ -126,7 +126,7 @@
 				$.loadMustache()
 			).then(function() {
 				messageTopic.autocomplete = messageTopic.input.autocomplete({
-					serviceUrl: wgServer + wgScript + '?action=ajax&rs=getLinkSuggest&format=json&nospecial=1&nsfilter=0',
+					serviceUrl: wgScript + '?action=ajax&rs=getLinkSuggest&format=json&nospecial=1&nsfilter=0',
 					onSelect: function(value, data) {
```
Comment (WikiaTech) [inline extensions/wikia/Wall/js/MessageTopic.js:130]: MAJOR Remove the unused function parameter "data". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

[project-id]: wikia:app

File: extensions/SemanticResultFormats/formats/Exhibit/exhibit/exhibit-api.js [hunk 6]
```diff
@@ -169,7 +169,7 @@
         }
 
         if (useLocalResources) {
-            Exhibit.urlPrefix = wgServer + wgScriptPath + "/extensions/SemanticResultFormats/formats/Exhibit/exhibit/";
+            Exhibit.urlPrefix = wgScriptPath + "/extensions/SemanticResultFormats/formats/Exhibit/exhibit/";
         }
 
         if (Exhibit.params.locale) { // ISO-639 language codes,
```
Comment (WikiaTech) [inline extensions/SemanticResultFormats/formats/Exhibit/exhibit/exhibit-api.js:175]: MAJOR Merge this if statement with the nested one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACollapsibleIfStatements)

[project-id]: wikia:app

File: extensions/SemanticResultFormats/formats/Exhibit/exhibit/extensions/time/time-extension.js [hunk 7]
```diff
@@ -45,7 +45,7 @@
 
     if (!("Timeline" in window)) {
         //scriptURLs.push("http://api.simile-widgets.org/timeline/2.3.1/timeline-api.js?bundle=true");
```
Comment (WikiaTech) [inline extensions/SemanticResultFormats/formats/Exhibit/exhibit/extensions/time/time-extension.js:47]: MAJOR Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)

[project-id]: wikia:app

File: extensions/SemanticResultFormats/formats/Exhibit/SRF_Exhibit.js [hunk 8]
```diff
@@ -102,7 +102,7 @@ function createExhibit() {//overload Exhibit's autoCreate-functionality
     for(var i = 0; i < ex_linkcounter; i++){ //lenses
```
Comment (WikiaTech) [inline extensions/SemanticResultFormats/formats/Exhibit/SRF_Exhibit.js:102]: MAJOR Rename "i" as this name is already used in declaration at line 82. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2814)

[project-id]: wikia:app

File: extensions/SemanticResultFormats/formats/Exhibit/SRF_Exhibit.js [hunk 9]
```diff
@@ -102,7 +102,7 @@ function createExhibit() {//overload Exhibit's autoCreate-functionality
...
 		var test = document.getElementById("linkcontent"+i);
```
Comment (WikiaTech) [inline extensions/SemanticResultFormats/formats/Exhibit/SRF_Exhibit.js:103]: MAJOR Rename "test" as this name is already used in declaration at line 92. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2814)

[project-id]: wikia:app

**Comment by WikiaTech** [comment]
SonarQube analysis reported 276 issues
* ![CRITICAL][CRITICAL] 5 critical
* ![MAJOR][MAJOR] 258 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 12 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [exhibit-api.js#L223](https://github.com/Wikia/app/blob/9c8974ee6dce209b5f9cdded18c105f1b97cf2c7/extensions/SemanticResultFormats/formats/Exhibit/exhibit/exhibit-api.js#L223): Review the arguments of this "eval" call to make sure they are validated. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEval)
1. ![CRITICAL][CRITICAL] [GameGuidesSponsored.js#L118](https://github.com/Wikia/app/blob/9c8974ee6dce209b5f9cdded18c105f1b97cf2c7/extensions/wikia/GameGuides/js/GameGuidesSponsored.js#L118): "addNew" declared at line 42 expects 1 argument, but 2 were provided. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS930)
1. ![CRITICAL][CRITICAL] [WMU.js#L455](https://github.com/Wikia/app/blob/9c8974ee6dce209b5f9cdded18c105f1b97cf2c7/extensions/wikia/WikiaMiniUpload/js/WMU.js#L455): TypeError can be thrown as "event" might be null or undefined here. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2259)
1. ![CRITICAL][CRITICAL] [WMU.js#L581](https://github.com/Wikia/app/blob/9c8974ee6dce209b5f9cdded18c105f1b97cf2c7/extensions/wikia/WikiaMiniUpload/js/WMU.js#L581): Review the arguments of this "eval" call to make sure they are validated. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEval)
1. ![CRITICAL][CRITICAL] [jquery.wikia.js#L83](https://github.com/Wikia/app/blob/9c8974ee6dce209b5f9cdded18c105f1b97cf2c7/resources/wikia/jquery.wikia.js#L83): Review the arguments of this "eval" call to make sure they are validated. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEval)
1. ![MAJOR][MAJOR] [SRF_Exhibit.js#L1](https://github.com/Wikia/app/blob/9c8974ee6dce209b5f9cdded18c105f1b97cf2c7/extensions/SemanticResultFormats/formats/Exhibit/SRF_Exhibit.js#L1): Function has a complexity of 21 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. ![MAJOR][MAJOR] [SRF_Exhibit.js#L13](https://github.com/Wikia/app/blob/9c8974ee6dce209b5f9cdded18c105f1b97cf2c7/extensions/SemanticResultFormats/formats/Exhibit/SRF_Exhibit.js#L13): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [SRF_Exhibit.js#L16](https://github.com/Wikia/app/blob/9c8974ee6dce209b5f9cdded18c105f1b97cf2c7/extensions/SemanticResultFormats/formats/Exhibit/SRF_Exhibit.js#L16): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)
1. ![MAJOR][MAJOR] [SRF_Exhibit.js#L20](https://github.com/Wikia/app/blob/9c8974ee6dce209b5f9cdded18c105f1b97cf2c7/extensions/SemanticResultFormats/formats/Exhibit/SRF_Exhibit.js#L20): Remove the declaration of the unused 'th' variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedVariable)
1. ![MAJOR][MAJOR] [SRF_Exhibit.js#L23](https://github.com/Wikia/app/blob/9c8974ee6dce209b5f9cdded18c105f1b97cf2c7/extensions/SemanticResultFormats/formats/Exhibit/SRF_Exhibit.js#L23): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

[project-id]: wikia:app
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 276 issues
* ![CRITICAL][CRITICAL] 5 critical
* ![MAJOR][MAJOR] 258 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 12 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [exhibit-api.js#L223](https://github.com/Wikia/app/blob/84c410c89987acec94e5a428a8d10e7cd55cf6e0/extensions/SemanticResultFormats/formats/Exhibit/exhibit/exhibit-api.js#L223): Review the arguments of this "eval" call to make sure they are validated. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEval)
1. ![CRITICAL][CRITICAL] [GameGuidesSponsored.js#L118](https://github.com/Wikia/app/blob/84c410c89987acec94e5a428a8d10e7cd55cf6e0/extensions/wikia/GameGuides/js/GameGuidesSponsored.js#L118): "addNew" declared at line 42 expects 1 argument, but 2 were provided. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS930)
1. ![CRITICAL][CRITICAL] [WMU.js#L455](https://github.com/Wikia/app/blob/84c410c89987acec94e5a428a8d10e7cd55cf6e0/extensions/wikia/WikiaMiniUpload/js/WMU.js#L455): TypeError can be thrown as "event" might be null or undefined here. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2259)
1. ![CRITICAL][CRITICAL] [WMU.js#L581](https://github.com/Wikia/app/blob/84c410c89987acec94e5a428a8d10e7cd55cf6e0/extensions/wikia/WikiaMiniUpload/js/WMU.js#L581): Review the arguments of this "eval" call to make sure they are validated. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEval)
1. ![CRITICAL][CRITICAL] [jquery.wikia.js#L83](https://github.com/Wikia/app/blob/84c410c89987acec94e5a428a8d10e7cd55cf6e0/resources/wikia/jquery.wikia.js#L83): Review the arguments of this "eval" call to make sure they are validated. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEval)
1. ![MAJOR][MAJOR] [SRF_Exhibit.js#L1](https://github.com/Wikia/app/blob/84c410c89987acec94e5a428a8d10e7cd55cf6e0/extensions/SemanticResultFormats/formats/Exhibit/SRF_Exhibit.js#L1): Function has a complexity of 21 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. ![MAJOR][MAJOR] [SRF_Exhibit.js#L13](https://github.com/Wikia/app/blob/84c410c89987acec94e5a428a8d10e7cd55cf6e0/extensions/SemanticResultFormats/formats/Exhibit/SRF_Exhibit.js#L13): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [SRF_Exhibit.js#L16](https://github.com/Wikia/app/blob/84c410c89987acec94e5a428a8d10e7cd55cf6e0/extensions/SemanticResultFormats/formats/Exhibit/SRF_Exhibit.js#L16): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)
1. ![MAJOR][MAJOR] [SRF_Exhibit.js#L20](https://github.com/Wikia/app/blob/84c410c89987acec94e5a428a8d10e7cd55cf6e0/extensions/SemanticResultFormats/formats/Exhibit/SRF_Exhibit.js#L20): Remove the declaration of the unused 'th' variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedVariable)
1. ![MAJOR][MAJOR] [SRF_Exhibit.js#L23](https://github.com/Wikia/app/blob/84c410c89987acec94e5a428a8d10e7cd55cf6e0/extensions/SemanticResultFormats/formats/Exhibit/SRF_Exhibit.js#L23): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

[project-id]: wikia:app
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by ludwikkazmierczak** [comment]
Tested and Accepted

**Pull request merged**